source/data-modeling/persistence-configuration.txt

```diff
@@ -0,0 +1,252 @@
+by the current thread.
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+  :language: ruby
+  :start-after: start store_in lambda example
+  :end-before: end store_in lambda example
+
+.. _mongoid-set-runtime-persistence-options:
+
+Setting Runtime Persistence Options
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can use the ``with()`` method on a model class or instance to change
+a model's persistence configuration for a group of operations during runtime.
+
+By default, the ``Band`` class will store documents in a collection called
+``bands``. By using the ``with()`` method, you can temporarily perform operations 
+on the ``Band`` model class's documents using a different client, database, or 
+collection.
+
+The following example shows how to perform operations on the ``Band`` model class
+within a different client, database, and collection.
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+  :language: ruby
+  :start-after: start with() example
+  :end-before: end with() example
+
+The persistence context for the model class or instance that ``with`` is called
+on is changed to the temporary persistence context defined by the arguments
+passed to the ``with()`` method within the block. The model class or instance 
+that ``with`` was called on is yielded to the block.
+
+The temporary persistence context applies to both queries and writes.
+
+Ensuring Consistency in Contexts
+++++++++++++++++++++++++++++++++
+
+If a document is saved in a temporary persistence context, it may not exist
+in the default persistence context. For example, if you insert a ``Band`` model
+document into the ``artists`` collection, the same document will not be inserted
+into the ``bands`` collection. The following example shows how to correctly 
+ensure consistency between contexts.
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+  :language: ruby
+  :start-after: start consistency example
+  :end-before: end consistency example
+
+.. important:: Using Block Scope
+   
+   As of {+odm+} 6.0, the ``with`` method must always be called with a block.
+   This is because a new client is created in the background with the
+   corresponding options that are passed to ``with``. The scope when this new
+   client is used must be well-defined so the client can be closed and
+   its resources can be freed.
+
+``with`` can also apply directly to read and write operations. The following 
+example reads and counts data from the secondary node in an application:
+
+.. literalinclude:: /includes/data-modeling/persistence-configuration.rb
+  :language: ruby
+  :start-after: start read configuration
+  :end-before: end read configuration
+
+Setting Persistence Contexts Globally
++++++++++++++++++++++++++++++++++++++
+
+{+odm+} provides the ability to globally define the persistence configuration at
+the database and client level. This is helpful if you want to switch the 
```

Comment:
S: explain what it means to globally define this configuration--does this apply to all operations on a particular database/client?